Add support for customBooleanFields to the IdentityX profile component

B77Mills opened the pull request:
**Will require: https://github.com/parameter1/identity-x/pull/6**

<img width="829" alt="Screen Shot 2022-01-05 at 11 35 10 AM" src="https://user-images.githubusercontent.com/3845869/148263987-5f2f1b09-c24e-4363-9056-a7a986cb3110.png">

B77Mills committed:
Add customBooleanField to fragment

B77Mills committed:
Add boolean field component

B77Mills committed:
Add customBoolean support to profile componet

B77Mills committed:
add mutation support for customBooleanFields

B77Mills committed:
cleanup unused variables

B77Mills commented:
![](https://github.trello.services/images/mini-trello-icon.png) [How to add checkbox fields to Identity-X](https://trello.com/c/adTAKA4F/258-how-to-add-checkbox-fields-to-identity-x)

zarathustra323 commented on packages/marko-web-identity-x/browser/profile.vue:
```diff
@@ -345,6 +371,10 @@ export default {
     },
 
+    onCustomBooleanChange() {
+      this.user.customBooleanFieldAnswers = this.customBooleanFieldAnswers;
```
I'm a bit confused by what's happening here... how is the change being applied if the array is just being reset to itself?

B77Mills replied:
@zarathustra323  I was missing a chunk of this from my site code.  Sorry it should take a answerId now and set the value to the opposite of what it was and also set the user.customBooleanFieldAnsers to this.customBooleanAnswers

B77Mills committed:
fix the onCustomBooleanChange function

On input change also flip the value related to the customBooleanAnswer to the opposite value & reset that on the current user.
